crates/parser2/src/parser/expr.rs

```diff
@@ -167,6 +175,10 @@ fn infix_binding_power<S: TokenStream>(parser: &mut Parser<S>) -> Option<(u8, u8
         Star | Slash | Percent => (130, 131),
         Star2 => (141, 140),
         Dot => (151, 150),
+        Eq => {
+            // `Assign` and `AugAssign` have the same binding power
+            (40, 41)
+        }
```

Comment:
This doesn't return the correct binding power if the current token is an augmented assign operator. You need to peek one to know if it's an assignment or not when the current token is arithmetic binary op.
e.g., the root expression for `1 + 2 *= 3` should be `(1 + 2) *= 3`, but the current implementation parses as `1 + (2 *= 3)`.